src/internal/active-elements.ts

```diff
@@ -20,3 +20,13 @@ export function* activeElements(activeElement: Element | null = document.activeE
     yield* activeElements(activeElement.shadowRoot.activeElement);
   }
 }
+
+export function getDeepestActiveElement() {
+  let activeElement;
+
+  for (const element of activeElements()) {
+    activeElement = element;
+  }
+
+  return activeElement;
```

Comment:
At a glance, it's not obvious that `activeElements()` is a generator so this loop looks strange. This can probably be simplified a bit:

```js
return [...activeElements()].pop()
```

Reply:
on second thought, the logic is annoying to do that, fixed!